File: GithubUserJava/app/src/main/java/com/example/myapplication/data/local/entity/FavoriteEntity.java
package com.example.myapplication.data.local.entity;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "favorite")
public class FavoriteEntity implements Parcelable {

    @PrimaryKey(autoGenerate = false)
    @ColumnInfo(name = "username")
    @NonNull
    private String username = "";

    @ColumnInfo(name = "avatar_url")
    private String avatarUrl;

    public FavoriteEntity() {
        // Default constructor required for Room
    }

    protected FavoriteEntity(Parcel in) {
        username = in.readString();
        avatarUrl = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(username);
        dest.writeString(avatarUrl);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<FavoriteEntity> CREATOR = new Creator<FavoriteEntity>() {
        @Override
        public FavoriteEntity createFromParcel(Parcel in) {
            return new FavoriteEntity(in);
        }

        @Override
        public FavoriteEntity[] newArray(int size) {
            return new FavoriteEntity[size];
        }
    };

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getAvatarUrl() {
        return avatarUrl;
    }

    public void setAvatarUrl(String avatarUrl) {
        this.avatarUrl = avatarUrl;
    }
}

File: GithubUserKotlin/app/src/main/java/com/example/myapplication/data/local/entity/FavoriteEntity.kt
package com.example.myapplication.data.local.entity

import android.os.Parcelable
import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.parcelize.Parcelize

@Parcelize
@Entity(tableName = "favorite")
data class FavoriteEntity(

    @PrimaryKey(autoGenerate = false)
    @ColumnInfo(name = "username")
    var username : String = "",

    @ColumnInfo(name = "avatar_url") 
    var avatarUrl : String? = null
) : Parcelable